### PR TITLE
Rules UI a11y updates

### PR DESCRIPTION
**Pull request by jamesjacobs**
This branch contains the following changes:

### Rules

**Event based rules**
- [Lexicon] Danger panel replaced with `form.error_summary` component on lexicon page
- [Lexicon] Table caption added to lexicon example rules table
- [Lexicon] Rules table updated with correct datatable responsive markup
- Remove link replaced with button with improved a11y label
- Rule block legend text updated to provide a better SR experience
- Error class moved from wrapping rule block `fieldset` to `.rule-block-wrapper` to keep error styling to the block and not bleed over unto And/Or buttons
- Rule block errors should now be passed to the individual fields and not passed via the `error` option on the block - this ensures error text is correctly associated to inputs and not orphaned. 
- Headings bumped to `h2` on rules blocks to avoid axe warnings

**Logic based rules**
- [Lexicon] Lexicon page added to rules patterns for logic based rules pattern
- [Lexicon] XFP rules page playground removed
- Form input markup updated to ensure labels are associated to inputs
- Rule blocks now use `fieldset` and `legend` rather than `div`'s to improve SR experience
- Rule block headings `span.rule-block__heading` changed to `h2`'s to improve SR navigation
- Rule block colour headings now have `font-weight: bold` to ensure coloured headings pass colour contrast - previously they did meet the wcag definition of large text but failed due to not being bold.

### Datatables
- Fixed an issue where collapsing datatables with `data-select="false"` would incorrectly show an empty first column (containing the hidden responsive toggle buttons).
- Fixed an issue where the first column was not sortable when `data-select="true"` was used.

Documentation update to event and logic examples to follow.

## Breaking change checklist

The following points should be considered as an early-warning of introducing a breaking change to a Continuum platform product. This is not an exhaustive list of what constitutes a breaking change.

| Does this pull request... | Yes / No |
| ---- | ---- |
| require changes to `main.js` | ✅  No |
| require changes to `index.js` | ✅  No |
| require changes to `pulsar.scss` | ✅  No |
| require changes to `gruntfile.js` | ✅  No |
| require changes to `package.json` (not including dev dependencies) | ✅  No |
| require changes to `base.html.twig` (or other core views like header/footer) |  ✅  No |
| require new arguments to be passed to a js component | ✅  No |
| require markup changes to maintain functionality | 🔴  YES (Numerous changes to rules views) |
| require changes to existing unit tests | ✅  No|

### CODE REVIEW

**Comment by jamesjacobs** [comment]
Fixes issues raised in #1180

File: stylesheets/_component.rules.scss [hunk 1]
```diff
@@ -172,15 +178,32 @@
...
 .conditional-block--flush {
     margin: 24px 0 0 58px;
+    position: relative;
 
     @include respond-min($screen-tablet) {
         margin: 24px 0 0 68px;
     }
+
+    &::after {
+        color: #dedede;
```
Comment (Stanton) [inline stylesheets/_component.rules.scss:188]: anything in the palette we can use?

File: stylesheets/_component.rules.scss [hunk 2]
```diff
@@ -172,15 +178,32 @@
...
+        content: '\f061';
+        font-family: 'FontAwesome';
+        font-size: 19px;
```
Comment (Stanton) [inline stylesheets/_component.rules.scss:191]: Will `$font-size-large` be enough?

File: stylesheets/_component.rules.scss [hunk 3]
```diff
@@ -189,13 +212,27 @@
...
         top: 0;
         width: 4px;
         z-index: 1;
     }
+
+    .btn {
+        margin-left: 25px;
+    }
+
+    &::after {
```
Comment (Stanton) [inline stylesheets/_component.rules.scss:227]: same var comments as above?

File: stylesheets/_component.rules.scss [hunk 4]
```diff
@@ -493,6 +535,7 @@ td.table__form .rules + .form__group:last-of-type {
     }
 
     .rule-block__heading {
+        font-weight: bold;
```
Comment (Stanton) [inline stylesheets/_component.rules.scss:538]: preferred over `$font-family-regular` ?

File: stylesheets/_component.rules.scss (hunk 2 above)
Comment (jamesjacobs) [reply stylesheets/_component.rules.scss]: Swapped

File: stylesheets/_component.rules.scss (hunk 3 above)
Comment (jamesjacobs) [reply stylesheets/_component.rules.scss:227]: Swapped

File: stylesheets/_component.rules.scss (hunk 4 above)
Comment (jamesjacobs) [reply stylesheets/_component.rules.scss:538]: Used to avoid contrast warnings in SI. Alternative would be to change logic bg colours.

**Pull request merged**